drivers/ec_host_cmd_periph/Kconfig

```diff
@@ -31,6 +35,55 @@ config EC_HOST_CMD_PERIPH_ESPI
 	  Enable support for Embedded Controller host commands using
 	  the eSPI bus.
 
+config EC_HOST_CMD_PERIPH_SHI
+	bool "Host commands support using SHI"
+	help
+	  Enable support for Embedded Controller host commands using
+	  the Serial Host Interface.
+
+endchoice
+
+if EC_HOST_CMD_PERIPH_SHI
+
+choice
+	prompt "SHI driver"
+	default EC_HOST_CMD_PERIPH_SHI_NPCX if SOC_FAMILY_NPCX
+	default EC_HOST_CMD_PERIPH_SHI_ITE if SOC_IT8XXX2
+
+config EC_HOST_CMD_PERIPH_SHI_NPCX
+	bool "SHI by Nuvoton"
+	depends on DT_HAS_NUVOTON_NPCX_SHI_ENABLED
+	help
+	  This option enables the driver for SHI peripheral in the
+	  Nuvoton NPCX chip.
+
+config EC_HOST_CMD_PERIPH_SHI_ITE
+	bool "SHI by ITE"
+	depends on DT_HAS_ITE_IT8XXX2_SHI_ENABLED
+	help
+	  This option enables the driver for SHI peripheral in the
+	  ITE IT8xxx2 chips family.
+
 endchoice
 
+config EC_HOST_CMD_PERIPH_SHI_MAX_REQUEST
+	hex "Max data size for the version 3 request packet"
+	default 0x220 if EC_HOST_CMD_PERIPH_SHI_NPCX
+	default 0x100 if EC_HOST_CMD_PERIPH_SHI_ITE
+	help
+	  This option indicates maximum data size for a version 3 request
+	  packet. This must be big enough to handle a request header of host
+	  command, flash write offset/size, and 512 bytes of flash data.
+
+config EC_HOST_CMD_PERIPH_SHI_MAX_RESPONSE
+	hex "Max data size for the version 3 response packet"
+	default 0x220 if EC_HOST_CMD_PERIPH_SHI_NPCX
+	default 0x100 if EC_HOST_CMD_PERIPH_SHI_ITE
```

Comment:
In ITE SHI, the maximum data size of response should not be greater than 248bytes, which needs to be added with EC_SPI_PREAMBLE_LENGTH + EC_SPI_PAST_END_LENGTH.
SPI_TX_MAX_FIFO_SIZE is not equal to EC_HOST_CMD_PERIPH_SHI_MAX_RESPONSE.

Reply:
Right, I've merged two defines into one. Thanks for noticing it and correcting me, I will change this.